python_tests/pytest_input_generators.py

```diff
@@ -395,13 +489,44 @@ def elementwise_binary_generator(
     # Typical inputs
     for shape in shapes:
         yield SampleInput(make_arg(shape), make_arg(shape))
-
-    # Noncontiguous inputs
-    for shape in shapes:
         yield SampleInput(
             make_arg(shape, noncontiguous=True), make_arg(shape, noncontiguous=True)
         )
 
+    if enable_broadcast_testing:
+        broadcast_shapes = (
+            ((1,), ()),
+            ((2,), ()),
+            ((1,), (2,)),
+            ((2, 1), (2,)),
+            ((1, 2), (2,)),
+            ((3, 2), (2,)),
+            ((1, 3, 2), (2,)),
+            ((1, 3, 2), (3, 2)),
+            ((3, 1, 2), (3, 2)),
+            ((2, 3, 2), ()),
+            ((3, 1, 2), (1, 3, 2)),
+        )
+        for lhs_shape, rhs_shape in broadcast_shapes:
+            yield SampleInput(make_arg(lhs_shape), make_arg(rhs_shape))
+            yield SampleInput(
+                make_arg(lhs_shape, noncontiguous=True),
+                make_arg(rhs_shape, noncontiguous=True),
+            )
+
+    if (
+        enable_large_value_testing
+        and dtype != torch.bool
+        and dtype not in complex_dtypes
+    ):
+        yield _special_value_binary_generator(_large_values, dtype, requires_grad)
+
+    if enable_small_value_testing and dtype != torch.bool:
+        yield _special_value_binary_generator(_small_values, dtype, requires_grad)
+
+    if enable_extremal_value_testing and dtype in float_complex_dtypes:
+        yield _special_value_binary_generator(_extremal_values, dtype, requires_grad)
+
```

Comment:
nitpick: we probably also wanted to see `_extremal_values` interacting with normal values in a binary op fashion.